### ISSUE TITLE
[Bug]: Dialog scroll on inner div is broken on high zoom

### ISSUE THREAD

**Issue by kumarritikms** [microsoft/fluentui#26973]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: macOS 13.2
    CPU: (10) x64 Apple M1 Pro
    Memory: 32.02 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
Browsers:
    Chrome: 110.0.5481.100
    Firefox: 101.0
    Safari: 16.3
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codepen.io/kumarritik/pen/LYJbzmZ

## Actual Behavior
On shifting the scroll from scrollableContent to inner div. The scroll is working fine in normal zoom(100%) but is broken for a zoom higher than 400% and on mobile devices.

Scroll at 100% zoom,
<img width="464" alt="image" src="https://user-images.githubusercontent.com/87644102/221503812-24ab4b01-c02e-4700-91c7-d498ebfb9ae0.png">

Scroll at 400% zoom (the scroll does not allow to scroll beyond a particular point),
<img width="1283" alt="image" src="https://user-images.githubusercontent.com/87644102/221504137-733ba19f-ec4a-4127-bac0-001a92edd4bb.png">

## Expected Behavior
The scroll should be able to scroll the content on all zoom levels and on mobile devices as well.

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
